manifests/init.pp

```diff
@@ -7,11 +11,7 @@
   String  $scripts_dir                 = "${install_dir}/scripts"
 ) {
 
-  if ( versioncmp('2017.3.0', $facts['pe_server_version']) <= 0 ) {
-    $psql_version = '9.6'
-  } else {
-    $psql_version = '9.4'
-  }
+  $psql_version = $facts['pe_postgresql_info']['installed_server_version']
```

Comment:
I was thinking the following might work if the fact does not resolve it could be an empty string instead of Undef. The `undef` should cause the backup class to fail validation.

```
$psql_version = $facts['pe_postgresql_info']['installed_server_version'] ? {
  undef   => undef,
  default => "${facts['pe_postgresql_info']['installed_server_version']}"
  }
```

Reply:
commit submitted with that